compiler_opt/rl/corpus.py

```diff
@@ -14,12 +14,106 @@
 # limitations under the License.
 """ModuleSpec definition and utility command line parsing functions."""
 
+from absl import logging
 from dataclasses import dataclass
+from typing import Iterable, Optional, Tuple, List
+
+import os
+import tensorflow as tf
 
 
 @dataclass(frozen=True)
 class ModuleSpec:
   """Dataclass describing an input module and its compilation command options.
   """
   name: str
-  has_thinlto: bool = False
+  _exec_cmd: Tuple[str, ...] = ()
+
+  def cmd(self) -> List[str]:
+    """Retrieves the compiler execution options.
+    """
+    # Using a getter as this will be modified in the future
+    return list(self._exec_cmd)
+
+
+def read(
```

Comment:
something like build_modulespecs_from_datapath, being more descriptive of what the function do.